Update install scripts

Keith-Khadar opened the pull request:
* The line 
```bash
python3 -m pip config set global.break-system-packages true
sudo apt install -y python3-vcstool || sudo pip3 install -U vcstool
``` 
did not seem work on my system so I switched it to 
```bash
sudo pip3 install -U vcstool --break-system-packages
```
* I also noticed that colcon was not automatically installed so I also added it 
```bash
# Install Colcon
sudo pip3 install -U colcon-common-extensions --break-system-packages
```
* Lastly I just added a ``-y`` to this line so that it doesnt prompt the user 
```bash
sudo apt-get install -y python3-rosdep
```

cbrxyz commented:
> The line
> ```py
> python3 -m pip config set global.break-system-packages true
> ```
> did not seem to work

What are your python and pip versions? The global config line is supposed to enable the `--break-system-packages` for all invocations of pip, so I'm weary to remove it.

The colcon/`-y` changes seem good though!

Keith-Khadar commented:
![image](https://github.com/user-attachments/assets/fd093038-9343-443e-a722-c5629ba87a5f)

I am not sure what ``python3 -m pip config set global.break-system-packages true`` does. It says that it wrote to pip.conf, but then when I use pip, I still get an externaly-managed-environment error.

Also the only other place we use pip, we pass in that ``--break-system-packages flag`` anyways (``sudo pip3 install -r requirements.txt --break-system-packages``).

cbrxyz commented:
What `python3 -m pip config set global.break-system-packages true` is doing is setting `break-system-packages = true` in your pip config file, which is telling `pip` that it's totally okay to install packages that could break the system. I agree that the `--break-system-packages` flags shouldn't be set in those individual `pip3 install` calls. I just removed them in 166f582 and [tested it](https://github.com/uf-mil/mil2/actions/runs/12451739222/job/34759958292) and it seems to have passed fine.

Maybe something is not right with your pip installation? Can you ensure that the flag was actually written to `~/.config/pip/pip.conf`, and if so, why your pip isn't reading from that config file? (maybe a path issue?)

Keith-Khadar commented:
Ok I got it. It was an issue with sudo. So apparently sudo doesn't preserve your current environment variables for the new process unless explicitly told to with --preserve-env / -E. So for me to run it I need to use 
``sudo -E pip3 install -U vcstool``

cbrxyz commented:
Why do we need sudo for a call to pip install?

On Sun, Dec 22, 2024 at 1:47 AM, Keith Khadar ***@***.***(mailto:On Sun, Dec 22, 2024 at 1:47 AM, Keith Khadar <<a href=)> wrote:

> Ok I got it. It was an issue with sudo. So apparently sudo doesn't preserve your current environment variables for the new process unless explicitly told to with --preserve-env / -E. So for me to run it I need to use
> sudo -E pip3 install -U vcstool
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/uf-mil/mil2/pull/3#issuecomment-2558352047), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/AMSRCUHKUENUBKKZMBCUHCT2GZN77AVCNFSM6AAAAABUBCAY4KVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNJYGM2TEMBUG4).
> You are receiving this because your review was requested.Message ID: ***@***.***>

Keith-Khadar commented:
Idk its how we have always had it. I'll remove it and see what happens

cbrxyz commented:
Thanks keith! if it goes well, I’ll approve it tomorrow!

On Sun, Dec 22, 2024 at 2:00 AM, Keith Khadar ***@***.***(mailto:On Sun, Dec 22, 2024 at 2:00 AM, Keith Khadar <<a href=)> wrote:

> Idk it's how we have always had it. I'll remove it and see what happens
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/uf-mil/mil2/pull/3#issuecomment-2558354209), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/AMSRCUCBKS5I5OVXKEKEVAT2GZPPZAVCNFSM6AAAAABUBCAY4KVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNJYGM2TIMRQHE).
> You are receiving this because your review was requested.Message ID: ***@***.***>

cbrxyz commented:
> ok so if we remove the sudo then we would have to add /home/user/.local/bin to the path

Let's just keep the sudo then. Can you push the changes that add the global pip config change back and I can approve?

> ok might be a little bit of a bad idea but what if we try using virtual environements

We can do that, but it will just add the overhead of maintaining a virtual environment onto what we are doing. The packages we're installing aren't likely to break the system (they haven't in the past), and if it happens, then we can cross the bridge of setting up venv stuff then.

cbrxyz reviewed (approved):
grazie

Keith-Khadar commented:
Ok I added sudo and ``python3 -m pip config set global.break-system-packages true`` back

I did make one change to that line however. I added sudo in front of it making it ``sudo python3 -m pip config set global.break-system-packages true`` That way we edit the root config. Which is the one we call when we use ``sudo pip install`` that way we dont have to use ``sudo -E pip install`` 

Let me know what you think.
